vector-im/element-android
# Voice Broadcast - Start record

## Type of change


## Content

Handle the recording side of the voice broadcast with play/pause/resume/stop actions
Voice messages are recorded in `mp4` for technical reasons (the Android MediaRecorder does not allow to record in multiple files in other formats).
For the moment, voice messages are sent to the room as separate events in relation to the voice broadcast state event. 
The aggregation of the events and the conversion to `ogg` will be done in dedicated PRs.

Note: common voice messages are still recorded in `ogg`

## Motivation and context

Continue #7127 

## Screenshots / GIFs

(there is transparency because of gif compression)

![voice_broadcast_recording](https://user-images.githubusercontent.com/11990514/195811635-05b4960f-2232-4312-a4e3-07df4a2a0946.gif)

## Tests

- Enable the voice broadcast feature flag
- Start a voice broadcast in a room
- Verify that voice messages are sent to the room
- Verify that the VB is correctly paused & resumed when playing with the buttons 
- Stop the VB and observe that a final voice message has been sent

## Tested devices

- OS version(s):

## Checklist
